backend/LexBoxApi/GraphQL/ProjectMutations.cs

```diff
@@ -119,22 +121,41 @@ public record BulkAddProjectMembersResult(List<UserProjectRole> AddedMembers, Li
         var existingUsers = await dbContext.Users.Include(u => u.Projects).Where(u => input.Usernames.Contains(u.Username) || input.Usernames.Contains(u.Email)).ToArrayAsync();
         var byUsername = existingUsers.Where(u => u.Username is not null).ToDictionary(u => u.Username!);
         var byEmail = existingUsers.Where(u => u.Email is not null).ToDictionary(u => u.Email!);
-        foreach (var username in input.Usernames)
+        foreach (var usernameOrEmail in input.Usernames)
         {
-            var user = byUsername.GetValueOrDefault(username) ?? byEmail.GetValueOrDefault(username);
+            var user = byUsername.GetValueOrDefault(usernameOrEmail) ?? byEmail.GetValueOrDefault(usernameOrEmail);
             if (user is null)
             {
                 var salt = Convert.ToHexString(RandomNumberGenerator.GetBytes(SHA1.HashSizeInBytes));
-                var isEmailAddress = username.Contains('@');
-                // TODO: In the future we'll want to allow usernames in the form "Real Name <email@example.com>" and extract the real name from them
-                // For now, just:
-                var name = username;
+                var isEmailAddress = usernameOrEmail.Contains('@');
+                string name;
+                string? email;
+                string? username;
+                if (isEmailAddress)
+                {
+                    try {
+                        var parsed = new MailAddress(usernameOrEmail);
+                        email = parsed.Address;
+                        username = null;
+                        name = parsed.DisplayName;
+                        if (string.IsNullOrEmpty(name)) name = email;
```

Comment:
How would you feel about this? I feel like the part before `@` is slightly more suitable as a display name.
```suggestion
                        if (string.IsNullOrEmpty(name)) name = email.Split("@")[0];
```

Reply:
Done in commit 35507c2d.